xonsh/environ.py

```diff
@@ -142,7 +142,10 @@ def xonshconfig(env):
     'XDG_CONFIG_HOME': os.path.expanduser(os.path.join('~', '.config')),
     'XDG_DATA_HOME': os.path.expanduser(os.path.join('~', '.local', 'share')),
     'XONSHCONFIG': xonshconfig,
-    'XONSHRC': os.path.expanduser('~/.xonshrc'),
+    'XONSHRC': ((os.path.join(os.environ['ALLUSERSPROFILE'],
+                              'Application Data', 'xonsh', 'xonshrc'),
+                os.path.expanduser('~/.xonshrc')) if ON_WINDOWS
+               else ('/etc/xonshrc', os.path.expanduser('~/.xonshrc'))), 
```

Comment:
It seems like `$XONSHRC` also needs an entry in the `DEFAULT_ENSURERS` dict that looks like:

```python
'XONSHRC': (is_env_path, str_to_env_path, env_path_to_str),
```

This will make this variable be able to be passed into / out of the process environment.